### PR TITLE
Codesign dylibs

### PR DESCRIPTION
**Pull request by Noarkhh**
closes https://github.com/membraneframework/membrane_core/issues/652

### CODE REVIEW

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:af25125030e85b72dfdfb9e6ac4baf7645a40c19

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:7d1680990257644c9d4d51f74146bd4d7ec126b1

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:d0a85259de0a307024401f379aa1bb65f6dbd252

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:c9584c10c717533013d49afe19ed183276e29205

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:d8e9e9aaf45a74c7b5f8e96b82f13e4fed02b87d

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:12d8b6d4f0ca84c7a950b26d7f0de59f7a70de90

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:84fe9f6b66e8e3635f4842e31ba024094576e9f0

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:083b4b59f6f179cb0a8ef03054bcfd7e9d925550

File: src/scripts/precompile_for_macos.sh [hunk 1]
```diff
@@ -10,13 +10,19 @@ brew install $PACKAGE_NAME
...
+for l in *.dylib
 do
-    install_name_tool -id "@rpath/$f" $f
-    cp -a "$(readlink $f)" ~/project/workspace/$artifact_name/lib
+    cp -a "$(readlink $l)" ~/project/workspace/$artifact_name/lib
+    f=~/project/workspace/$artifact_name/lib/$l
+    if [ ! -L $f ]
+    then
+        install_name_tool -id "@rpath/$(basename $f)" $f
+        otool -L $f | tail -n +3 | awk -F" " '{print $1}' | ( grep "^${brew_prefix}" || [ "$?" == "1" ] ) | while read -r line; do install_name_tool -change $line "@rpath/$(basename $line)" $f; done
```
Comment (mat-hek) [inline src/scripts/precompile_for_macos.sh:25]: beautiful 😍 hope it works

File: src/scripts/precompile_for_macos.sh [hunk 2]
```diff
@@ -10,13 +10,19 @@ brew install $PACKAGE_NAME
...
+        [[ $ARCHITECTURE == arm ]] && codesign --sign - --force --preserve-metadata=entitlements,requirements,flags,runtime $f
+    fi
```
Comment (mat-hek) [inline src/scripts/precompile_for_macos.sh:27]: maybe let's add some comments and/or print what you do

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:451c84a5b2af13c97454bf591d6ebc5e9501844f

**Review by mat-hek — APPROVED** [review]

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:cdd1f556fc53633228ca2fa417edfdde13fb40bd

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:cbeb8e1dcc1b09ffd227cb607fa7f7903d609a9e

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:cbeb8e1dcc1b09ffd227cb607fa7f7903d609a9e

**Comment by bblaszkow06** [comment]
Your orb has been published to the CircleCI Orb Registry.
You can view your published orb on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=1.4.1

**Pull request merged**